src/Microsoft.IdentityModel.Tokens/BaseConfigurationComparer.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    internal class BaseConfigurationComparer : IEqualityComparer<BaseConfiguration>
+    {
+        public bool Equals(BaseConfiguration config1, BaseConfiguration config2)
+        {
+            if (config1 == null && config2 == null)
+                return true;
+            else if (config1 == null || config2 == null)
+                return false;
+            else if (config1.Issuer == config2.Issuer && config1.SigningKeys.Count == config2.SigningKeys.Count
+                     && !config1.SigningKeys.Select(x => x.InternalId).Except(config1.SigningKeys.Select(x => x.InternalId)).Any())
```

Comment:
Is the InternalId some generated hash of the crypto artifact? How can we be sure this isn't duplicated with a valid and invalid artifact in config1 vs config2?

Reply:
Is the InternalId some generated hash of the crypto artifact?
Yes

Reply:
The InternalId is based off the this spec: https://www.rfc-editor.org/rfc/rfc7638